src/rdkafka_telemetry.c

```diff
@@ -93,6 +94,14 @@ void rd_kafka_telemetry_clear(rd_kafka_t *rk,
                 rk->rk_telemetry.matched_metrics       = NULL;
                 rk->rk_telemetry.matched_metrics_cnt   = 0;
         }
+        TAILQ_FOREACH(rkb, &rk->rk_brokers, rkb_link) {
```

Comment:
nit: I think we should have functions in the same place for clearing (currently here) resetting (currently in reset_historical_metrics) and initializing (currently in rd_kafka_set_telemetry_broker_maybe).
Not a must-do but it'd make reading it easier

Reply:
Makes sense, I've done it in the later PR https://github.com/confluentinc/librdkafka/pull/4529 so can keep as it is here and it should be updated once we merge 4529.